src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -54,137 +115,853 @@ static ColliderCylinderInit D_80B90E54 = {
+
+void func_80B90010(ObjBean* this) {
+    this->actionFunc = func_80B90050;
+    ObjBean_SetDrawMode(this, BEAN_STATE_DRAW_LEAVES | BEAN_STATE_DRAW_SOIL);
+    this->unk_1B6.x = 0;
+    this->unk_1B6.y = 0xBB8;
+}
+
+// Control is returned to the player and the leaves start to flatten out
+void func_80B90050(ObjBean* this, GlobalContext* globalCtx) {
+    s16 temp_a0;
+    f32 temp_f2;
+
+    this->unk_1B6.x += 0x3E80;
+    this->unk_1B6.y += -0xC8;
+    temp_a0 = 6372.0f - Math_SinS(this->unk_1B6.x) * this->unk_1B6.y;
+
+    this->dyna.actor.scale.y = Math_SinS(temp_a0) * 0.17434467f;
+    this->dyna.actor.scale.x = this->dyna.actor.scale.z = Math_CosS(temp_a0) * 0.12207746f;
+    if (this->unk_1B6.y < 0) {
+        ObjBean_SetupWaitForWater(this);
+    }
+}
+
+void ObjBean_SetupWaitForWater(ObjBean* this) {
+    this->actionFunc = ObjBean_WaitForWater;
+    ObjBean_SetDrawMode(this, BEAN_STATE_DRAW_LEAVES | BEAN_STATE_DRAW_SOIL);
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    ObjBean_SetupLeavesStill(this);
+}
+
+#ifdef NON_MATCHING
+void ObjBean_WaitForWater(ObjBean* this, GlobalContext* globalCtx) {
+    this->transformFunc(this);
+    if (!(this->stateFlags & BEAN_STATE_BEEN_WATERED) && Flags_GetEnv(globalCtx, 5) && (D_80B90E30 == NULL) &&
+        (this->dyna.actor.xzDistToPlayer < 50.0f)) {
+        ObjBean_SetupGrowWaterPhase1(this);
+        D_80B90E30 = this;
+        OnePointCutscene_Init(globalCtx, 2210, -99, &this->dyna.actor, MAIN_CAM);
+        this->dyna.actor.flags |= 0x10;
+        return;
+    }
+    if ((D_80B90E30 == this) && !Flags_GetEnv(globalCtx, 5)) {
+        D_80B90E30 = NULL;
+    }
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bean/ObjBean_WaitForWater.s")
+#endif
+
+void ObjBean_SetupGrowWaterPhase1(ObjBean* this) {
+    this->actionFunc = ObjBean_GrowWaterPhase1;
+    ObjBean_SetDrawMode(this, BEAN_STATE_DRAW_LEAVES | BEAN_STATE_DRAW_SOIL);
+    ObjBean_SetupShakeLeavesFast(this);
+    this->timer = 50;
+}
+
+// Camera moves and leaves move quickly
+void ObjBean_GrowWaterPhase1(ObjBean* this, GlobalContext* globalCtx) {
+    this->transformFunc(this);
+    if (this->timer <= 0) {
+        ObjBean_SetupGrowWaterPhase2(this);
+    }
+}
+
+void ObjBean_SetupGrowWaterPhase2(ObjBean* this) {
+    this->actionFunc = ObjBean_GrowWaterPhase2;
+    ObjBean_SetDrawMode(this, BEAN_STATE_DRAW_SOIL | BEAN_STATE_DRAW_LEAVES | BEAN_STATE_DRAW_STALK);
+    this->stalkSizeMultiplier = 0.0f;
+}
+
+// BeanStalk is visable and is growing
+void ObjBean_GrowWaterPhase2(ObjBean* this, GlobalContext* globalCtx) {
+    this->transformFunc(this);
+    this->stalkSizeMultiplier += 0.001f;
+    this->dyna.actor.shape.rot.y = this->dyna.actor.home.rot.y + (s16)(this->stalkSizeMultiplier * 700000.0f);
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + this->stalkSizeMultiplier * 800.0f;
+    if (this->stalkSizeMultiplier >= 0.1f) { // 100 Frames
+        ObjBean_SetupGrowWaterPhase3(this);
+    }
```

Comment:
this comment doesnt make sense to me in this context

Reply:
How so? It functionally doubles as both the scale and a timer.

Reply:
Ah ok. I'm not as familiar with the code so I could be wrong. But it seems the fact that it takes 100 frames is just a consequence of the size they wanted it to be. Thinking about it as number of frames seems off to me
If the clarification is needed something like "it takes 100 frames to reach the specified size" would be more clear, but I'm leaning more toward it not being needed at all